cli/src/commands/git/remote/set_url.rs

```diff
@@ -42,7 +56,29 @@ pub fn cmd_git_remote_set_url(
     args: &GitRemoteSetUrlArgs,
 ) -> Result<(), CommandError> {
     let workspace_command = command.workspace_helper(ui)?;
-    let url = absolute_git_url(command.cwd(), &args.url)?;
-    git::set_remote_url(workspace_command.repo().store(), &args.remote, &url)?;
+
+    let fetch_url = match args.url {
+        Some(ref url) => process_url(command, Some(url))?,
+        None => process_url(command, args.fetch.as_ref())?,
+    };
+    let push_url = process_url(command, args.push.as_ref())?;
+
+    git::set_remote_urls(
+        workspace_command.repo().store(),
+        &args.remote,
+        fetch_url.as_deref(),
+        push_url.as_deref(),
+    )?;
     Ok(())
 }
+
+fn process_url(
+    command: &CommandHelper,
+    url: Option<&String>,
+) -> Result<Option<String>, CommandError> {
+    let url = match url {
+        Some(u) => Some(absolute_git_url(command.cwd(), u)?),
+        None => None,
+    };
```

Comment:
nit: at your option, `url.map(|u| absolute_git_url(..)).transpose()`.

I think it's also good to add closure `let process_url = |url| ...;` instead of free function.